prov/gni/src/gnix_msg.c

```diff
@@ -314,27 +411,28 @@ static void __gnix_msg_copy_unaligned_get_data(struct gnix_fab_req *req)
 	int head_off, head_len, tail_len;
 	void *addr;
 
-	head_off = req->msg.send_addr & GNI_READ_ALIGN_MASK;
+	head_off = req->msg.send_info[0].send_addr & GNI_READ_ALIGN_MASK;
 	head_len = head_off ? GNI_READ_ALIGN - head_off : 0;
-	tail_len = (req->msg.send_addr + req->msg.send_len) &
+	tail_len = (req->msg.send_info[0].send_addr + req->msg.send_info[0].send_len) &
 			GNI_READ_ALIGN_MASK;
 
 	if (head_off) {
 		addr = (uint8_t *)&req->msg.rndzv_head + head_off;
 
 		GNIX_INFO(FI_LOG_EP_DATA,
-			  "writing %d bytes to head (%p, 0x%x)\n",
-			  head_len, req->msg.recv_addr, *(uint32_t *)addr);
-		memcpy((void *)req->msg.recv_addr, addr, head_len);
+			  "writing %d bytes to head (%p, %hxx)\n",
```

Comment:
I know this was here before this commit, but I think this should be a GNIX_DEBUG.